### PR TITLE
Update WR (new texture cache allocator, shader deinit path).

### PR DESCRIPTION
**Pull request by glennw**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18096)

### CODE REVIEW

**Comment by jdm** [comment]
```
error[E0425]: cannot find value `browser` in this scope
   --> main.rs:186:5
    |
186 |     browser.browser.deinit();
    |     ^^^^^^^ not found in this scope

error: aborting due to previous error
```

File: components/compositing/compositor.rs [hunk 1]
```diff
@@ -340,6 +340,7 @@ impl webrender_api::RenderDispatcher for CompositorThreadDispatcher {
     }
 }
 
+
```
Comment (jdm) [inline components/compositing/compositor.rs]: nit: remove this.

**Review by jdm — APPROVED** [review]
I don't understand why that error is occurring, but otherwise this looks fine.

**Comment by glennw** [comment]
Rebased.

**Comment by jdm** [comment]
```
  ▶ FAIL [expected PASS] /_mozilla/mozilla/webgl/tex_image_2d_mipmap.html
  └   → /_mozilla/mozilla/webgl/tex_image_2d_mipmap.html 4b4de7a92a4ad4f71d2fe6f7c5609502607ffd7b
/_mozilla/mozilla/webgl/tex_image_2d_simple_ref.html 06711e6d8ba9520a7c0476d41130022d3e609160
Testing 4b4de7a92a4ad4f71d2fe6f7c5609502607ffd7b == 06711e6d8ba9520a7c0476d41130022d3e609160

  ▶ FAIL [expected PASS] /_mozilla/mozilla/webgl/tex_image_2d_abv.html
  └   → /_mozilla/mozilla/webgl/tex_image_2d_abv.html a9a5c967d45ac416acbdb47a6930a2ddf6771ff4
/_mozilla/mozilla/webgl/tex_image_2d_abv_ref.html d83355c858358b2d507f0d58d54f1efdc20e4be1
Testing a9a5c967d45ac416acbdb47a6930a2ddf6771ff4 == d83355c858358b2d507f0d58d54f1efdc20e4be1

  ▶ FAIL [expected PASS] /_mozilla/mozilla/webgl/clearcolor.html
  └   → /_mozilla/mozilla/webgl/clearcolor.html 2b3b81dee2e799bb62c5b337eedd6c91da918772
/_mozilla/mozilla/webgl/clearcolor_ref.html a83005ec2d5b8d21ca8536830ed9cb073c11545f
Testing 2b3b81dee2e799bb62c5b337eedd6c91da918772 == a83005ec2d5b8d21ca8536830ed9cb073c11545f
```

**Comment by glennw** [comment]
This is waiting on https://github.com/servo/webrender/pull/1581.

**Comment by glennw** [comment]
This should *in theory* be ready to go now...

**Pull request merged**